pkg/stanza/adapter/integration_test.go

```diff
@@ -151,7 +151,10 @@ func TestEmitterToConsumer(t *testing.T) {
 
 	err = logsReceiver.Start(context.Background(), componenttest.NewNopHost())
 	require.NoError(t, err)
-	defer func() { require.NoError(t, logsReceiver.Shutdown(context.Background())) }()
+	defer func() {
+		require.NoError(t, logsReceiver.emitter.Stop())
```

Comment:
Note: This is now necessary, because the receiver is not allowed to stop until the emitter does. In an actual pipeline, the emitter itself is owned by the pipeline, and the pipeline is responsible for closing it. Here, we need to do it manually.